The FDMA platform device memory resource was one byte too large.

Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
Index: linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stb7100.c
===================================================================
--- linux-2.6.23-stm.orig/arch/sh/kernel/cpu/sh4/setup-stb7100.c
+++ linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stb7100.c
@@ -279,7 +279,7 @@ static struct platform_device fdma_710x_
 	.resource = (struct resource[2]) {
 		[0] = {
 			.start = STB7100_FDMA_BASE,
-			.end   = STB7100_FDMA_BASE + 0x10000,
+			.end   = STB7100_FDMA_BASE + 0xffff,
 			.flags = IORESOURCE_MEM,
 		},
 		[1] = {
Index: linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stx7200.c
===================================================================
--- linux-2.6.23-stm.orig/arch/sh/kernel/cpu/sh4/setup-stx7200.c
+++ linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stx7200.c
@@ -446,7 +446,7 @@ static struct platform_device fdma0_7200
 	.resource = (struct resource[2]) {
 		[0] = {
 			.start = STB7200_FDMA0_BASE,
-			.end   = STB7200_FDMA0_BASE + 0x10000,
+			.end   = STB7200_FDMA0_BASE + 0xffff,
 			.flags = IORESOURCE_MEM,
 		},
 		[1] = {
@@ -466,7 +466,7 @@ static struct platform_device fdma1_7200
 	.resource = (struct resource[2]) {
 		[0] = {
 			.start = STB7200_FDMA1_BASE,
-			.end   = STB7200_FDMA1_BASE + 0x10000,
+			.end   = STB7200_FDMA1_BASE + 0xffff,
 			.flags = IORESOURCE_MEM,
 		},
 		[1] = {
@@ -487,7 +487,7 @@ static struct platform_device fdma_xbar_
 	.resource	= (struct resource[1]) {
 		{
 			.start	= STB7200_XBAR_BASE,
-			.end	= STB7200_XBAR_BASE+(4*1024),
+			.end	= STB7200_XBAR_BASE+(4*1024)-1,
 			.flags	= IORESOURCE_MEM,
 		},
 	},
